server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1093,13 +1098,22 @@ void runRandomly() {
                 }
 
                 try {
-                    if (rarely()) {
+                    if (randomBoolean() && randomBoolean() && randomBoolean()) {
```

Comment:
Running all tests in the class, 75-80% of the linearizability checks are against histories where all invocations come before all responses (form 1). Around half of the remaining checks are against histories with form (invocation*, response*, invocation*, response*), ie. two rounds of invocation/response (form 2). Only checking histories larger than size 100, the numbers are 63-65% of form 1 and 77- 80% of form 2. Larger than size 1000, 1 out of 2 histories were of form 1.

I believe form 1 will only fail if responses contains values that are not part of one of the invocations, but otherwise the linearizability checks will not detect any problems.

Perhaps we need to tweak the randomness to better provoke situations that could reveal linearizability problems?

Reply:
Thank you for the investigation. Tweaking the randomness here is not too straightforward, and has a more general impact on how these Coordinator tests run. I will leave this to a follow-up PR, as I don't want to any longer delay merging the infrastructure in this PR so that it is available to other follow-up work.